api/room/models.py

```diff
@@ -59,3 +59,11 @@ def receive_after_flush(session, context):
         if not verify_calendar_id(target.calendar_id):
             raise GraphQLError("Room calendar Id is invalid")
         pass
+
+
+cascade_soft_delete(
+    Room, 'room_resource', 'room_id'
+)
+cascade_soft_delete(
+    Room, 'events', 'room_id'
+)
```

Comment:
Kindly include devices and responses in soft-deletion of room responses.

Reply:
Responses and devices have no state column in their models so calling this function on them would not work.

As discussed in the backend-sync, stories are to be raised to fix this.